modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/IgniteH2Indexing.java

```diff
@@ -1979,6 +1986,26 @@ public QueryMemoryManager memoryManager() {
             U.warn(log, "Custom H2 serialization is already configured, will override.");
 
         JdbcUtils.serializer = h2Serializer();
+
+        cleanSpillDirectory();
+    }
+
+    /**
+     * Cleans spill directory. Spill directory is used for disk
+     * offloading of the intermediate results of heavy queries.
+     */
+    private void cleanSpillDirectory() {
+        try {
+            File spillDir = U.resolveWorkDirectory(
+                ctx.config().getWorkDirectory(),
+                DISK_SPILL_DIR,
+                false);
+
+            U.delete(spillDir);
+        }
+        catch (Exception e) {
+            log.debug("Failed to delete spill directory.", X.getFullStackTrace(e));
```

Comment:
Logging this at warning level may be useful for user in case an app has no rights to write to spill directory.
Also let's change msg to smth "Failed to cleanup SQL temporary directory with intermediate SQL query results from previous node run."

Reply:
Done